sqlmypy.py

```diff
@@ -141,14 +142,23 @@ def decl_info_hook(ctx):
 
         Base = declarative_base()
     """
+    cls_instances = []  # type: List[Instance]
+
+    if 'cls' in ctx.call.arg_names:
+        declarative_base_cls_arg = ctx.call.args[ctx.call.arg_names.index("cls")]
+        if isinstance(declarative_base_cls_arg, TupleExpr):
+            cls_instances = [Instance(item.node, []) for item in declarative_base_cls_arg.items]
+        else:
+            cls_instances = [Instance(declarative_base_cls_arg.node, [])]
+
     class_def = ClassDef(ctx.name, Block([]))
     class_def.fullname = ctx.api.qualified_name(ctx.name)
 
     info = TypeInfo(SymbolTable(), class_def, ctx.api.cur_mod_id)
     class_def.info = info
     obj = ctx.api.builtin_type('builtins.object')
-    info.mro = [info, obj.type]
-    info.bases = [obj]
+    info.bases = cls_instances + [obj]
+    calculate_mro(info)
```

Comment:
`calculate_mro()` can raise `MroError`, so must be wrapped in try-except block, converting the exception to an error message. I propose to make it non-blocking, erase base classes (except object), and set `fallback_to_any`.